storage/journal/src/main/java/io/atomix/storage/journal/SegmentedJournal.java

```diff
@@ -553,7 +553,7 @@ void resetHead(long index) {
    */
   void resetTail(long index) {
     for (SegmentedJournalReader<E> reader : readers) {
-      if (reader.getNextIndex() > index) {
+      if (reader.getNextIndex() >= index) {
```

Comment:
Log reader’s need to be reset when the reset index is the last index to ensure the next entry is decoded again.